packages/airnode-abi/src/decoding.ts

```diff
@@ -63,6 +73,10 @@ export function decode(encodedData: string): DecodedMap {
   // exactly what you got from the contract, including the header.
   const decodedData = ethers.utils.defaultAbiCoder.decode(decodingTypes, encodedData);
 
+  if (!reEncodedLengthMatches(decodingTypes, decodedData, encodedData.length)) {
```

Comment:
Why don't we compare the encoded data directly (but only its length)?

Reply:
It seemed that comparing the length would be enough based on [this](https://api3dao.atlassian.net/browse/AN-469?focusedCommentId=13795), but would you prefer to compare the string?

Reply:
I would require an exact match. This is a stricter requirement then checking length (but is the same in practice).

Also, should we say `Trailing bytes in encoded data are not allowed`? This assumes that the length of the newly encoded data is shorter (which is true, but it's not apparent from reading the code).

Reply:
@bbenligiray Would you have any input on this? Should we just compare length here or the entire string?

Reply:
Assuming that the ethers implementation is correct, we will always have an exact match if the lengths are identical, and the only reason the lengths won't match is if there are trailing bytes. So if we were checking if there is an exact match in addition to a length match, we would be validating the ethers implementation.

I don't see anything wrong with requiring an exact match though, but we would have to have a more general error message. We never expect to have trailing bytes either, so we probably don't need a specialized error message for that specifically.